docs/usage/frameworks/litestar.rst

```diff
@@ -327,20 +330,81 @@ Dependency injection is not available in middleware. Instead, you can create a n
     )
     alchemy = SQLAlchemyPlugin(config=sqlalchemy_config)
 
-    def middleware_factory(app: ASGIApp) -> ASGIApp:
-        async def my_middleware(scope: Scope, receive: Receive, send: Send) -> None:
-            # NOTE: You can also access the app state from `ASGIConnection`.
-            db_session = await alchemy.provide_session(scope["app"].state, scope)
+
+    async def my_guard(connection: ASGIConnection, _: BaseRouteHandler) -> None:
+        db_session = sqlalchemy_config.provide_session(connection.app.state, connection.scope)
+        # Access the database session here.
+
+
+    @get("/", guards=[my_guard])
+    async def hello() -> str:
+        return "Hello, world!"
+
+
+    app = Litestar(
+        route_handlers=[hello],
+        plugins=[alchemy],
+    )
+
+- ``get_session`` creates a new session instance every time it is called.
+
+.. code-block:: python
+
+    from litestar import Litestar, get
+    from litestar.plugins.sqlalchemy import (
+        AsyncSessionConfig,
+        SQLAlchemyAsyncConfig,
+        SQLAlchemyPlugin,
+    )
+    from litestar_saq import CronJob, QueueConfig, SAQConfig, SAQPlugin
+    from saq.types import Context
+
+    session_config = AsyncSessionConfig(expire_on_commit=False)
+    sqlalchemy_config = SQLAlchemyAsyncConfig(
+        connection_string="sqlite+aiosqlite:///test.sqlite",
+        before_send_handler="autocommit",
+        session_config=session_config,
+        create_all=True,
+    )
+    alchemy = SQLAlchemyPlugin(config=sqlalchemy_config)
+
+
+    async def background_task(_: Context) -> None:
+        async with sqlalchemy_config.get_session() as db_session:
             # Access the database session here.
-            await db_session.close()
-            ...
-            await app(scope, receive, send)
-    return my_middleware
+            pass
+
+
+    saq = SAQPlugin(
+        config=SAQConfig(
+            web_enabled=True,
+            use_server_lifespan=True,
+            queue_configs=[
+                QueueConfig(
+                    dsn="redis://localhost:6397/0",
+                    name="process_background_tasks",
+                    scheduled_tasks=[
+                        CronJob(
+                            function=background_task,
+                            cron="* * * * *",
+                            timeout=600,
+                            ttl=2000,
+                        ),
+                    ],
+                ),
+            ],
+        ),
+    )
+
+
+    @get("/")
+    async def hello() -> str:
+        return "Hello, world!"
```

Comment:
I don't think we should reference other plugins in our examples

Reply:
Take a look at swapping this out with a CLI plugin example.  

Great work so far though!